core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -137,9 +137,23 @@ public static List<String> manifestListLocations(Table table, Set<Long> snapshot
    * @return the location of statistics files
    */
   public static List<String> statisticsFilesLocations(Table table) {
+    return statisticsFilesLocations(table, null);
+  }
+
+  /**
+   * Returns locations of statistics files matching the given snapshotIds in a table.
+   *
+   * @param table table for which statistics files needs to be listed
+   * @param snapshotIds ids of snapshots for which statistics files will be returned. When null,
+   *     returns location of all the statistics files in a table.
+   * @return the location of statistics files
+   */
+  public static List<String> statisticsFilesLocations(Table table, Set<Long> snapshotIds) {
```

Comment:
can we simplify the logic of the original method using stream()? Also for this method, I think we can provide a more generic util method that takes in a filter, instead of just adding a specific method to filter by snapshot ID set

Reply:
Updated now. 
Originally I wanted to keep the same style as `manifestListLocations` and others. So, I didn't used modified.